pkgs/applications/audio/netease-cloud-music-gtk/default.nix

```diff
@@ -1,43 +1,60 @@
 { lib
 , stdenv
+, fetchFromGitHub
+, rustPlatform
+, meson
+, ninja
+, pkg-config
 , glib
-, gtk3
-, curl
-, dbus
+, gtk4
+, appstream-glib
+, desktop-file-utils
+, libxml2
+, wrapGAppsHook4
 , openssl
+, dbus
+, libadwaita
 , gst_all_1
-, pkg-config
-, rustPlatform
-, wrapGAppsHook
-, fetchurl
-, fetchFromGitHub
 }:
-rustPlatform.buildRustPackage rec {
+
+stdenv.mkDerivation rec {
   pname = "netease-cloud-music-gtk";
-  version = "1.2.2";
+  version = "2.0.1";
+
   src = fetchFromGitHub {
     owner = "gmg137";
-    repo = "netease-cloud-music-gtk";
+    repo = pname;
     rev = version;
-    sha256 = "sha256-42MaylfG5LY+TiYHWQMoh9CiVLShKXSBpMrxdWhujow=";
+    hash = "sha256-dlJZvmfw9+cavAysxVzCekgPdygg5zbU3ZR5BOjPk08=";
+  };
+
+  patches = [ ./cargo-lock.patch ];
```

Comment:
Is it better to let the upstream provide `Cargo.lock`?

Reply:
Upstream refused to do so: https://github.com/gmg137/netease-cloud-music-gtk/issues/118